src/main/haskell/kore/src/Kore/Step/Function/Matcher.hs

```diff
@@ -471,15 +472,19 @@ matchNonVarToPattern tools substitutionSimplifier first second
             -- predicates.
             -- TODO: Make a function for this.
             leftVars = freePureVariables first
+            rawSubstitution = Substitution.unwrap substitution
             (leftSubst, rightSubst) =
-                List.partition ((`elem` leftVars) . fst) substitution
+                List.partition ((`elem` leftVars) . fst) rawSubstitution
             finalPredicate = Predicated.toPredicate
-                Predicated { term = (), predicate, substitution = rightSubst }
+                Predicated
+                    { term = ()
+                    , predicate
+                    , substitution = Substitution.wrap rightSubst }
```

Comment:
I think our style guide would demand that the closing brace go on the next line:
```suggestion
                    , substitution = Substitution.wrap rightSubst
                    }
```